src/compositemark/errorbar.ts

```diff
@@ -1,42 +1,395 @@
+
+  function errorBarSinglePointPartMarksSpec(partName: ErrorBarPart, startingPosition: string = center, endingPosition:string = undefined) {
+    const markProperties: boolean | object = config.errorbar[partName];
+    const defaultMarkDef: Mark | MarkDef = (isBoolean(markProperties)) ? getMarkNameOfErrorBarPart(partName) : {
+      type: getMarkNameOfErrorBarPart(partName),
+      ...markProperties
+    };
+
+    return partLayerMixins<ErrorBarPartsMixins>(
+      markDef, partName, config.errorbar,
+      {
+        mark: defaultMarkDef,
+        encoding: {
+          [continuousAxis]: {
+            field: startingPosition + '_' + continuousAxisChannelDef.field,
+            type: continuousAxisChannelDef.type,
+            ...(scale ? {scale} : {}),
+            ...(axis ? {axis} : {})
+          },
+          ...(isString(endingPosition) ? {
+            [continuousAxis + '2']: {
+              field: endingPosition + '_' + continuousAxisChannelDef.field,
+              type: continuousAxisChannelDef.type
+            }
+          } : {}),
+          ...encodingWithoutSizeAndContinuousAxis
+        }
+      }
+    );
   }
 
   return {
     ...outerSpec,
+    transform,
     layer: [
+      ...errorBarSinglePointPartMarksSpec('bar'),
+      ...errorBarSinglePointPartMarksSpec('line'),
+      ...errorBarSinglePointPartMarksSpec('ticks', 'lower_rule'),
+      ...errorBarSinglePointPartMarksSpec('ticks', 'upper_rule'),
+      ...errorBarSinglePointPartMarksSpec('rule', 'lower_rule', 'upper_rule'),
+      ...errorBarSinglePointPartMarksSpec('point')
+    ]
+  };
+}
+
+function errorBarOrient(spec: GenericUnitSpec<Encoding<Field>, ErrorBar | ErrorBarDef>): Orient {
+  const {mark: mark, encoding: encoding, projection: _p, ..._outerSpec} = spec;
+
+  if (isFieldDef(encoding.x) && isContinuous(encoding.x)) {
+    // x is continuous
+    if (isFieldDef(encoding.y) && isContinuous(encoding.y)) {
+      // both x and y are continuous
+      if (encoding.x.aggregate === undefined && encoding.y.aggregate === ERRORBAR) {
+        return 'vertical';
+      } else if (encoding.y.aggregate === undefined && encoding.x.aggregate === ERRORBAR) {
+        return 'horizontal';
+      } else if (encoding.x.aggregate === ERRORBAR && encoding.y.aggregate === ERRORBAR) {
+        throw new Error('Both x and y cannot have aggregate');
+      } else {
+        if (isMarkDef(mark) && mark.orient) {
+          return mark.orient;
+        }
+
+        // default orientation = vertical
+        return 'vertical';
+      }
+    }
+
+    // x is continuous but y is not
+    return 'horizontal';
+  } else if (isFieldDef(encoding.y) && isContinuous(encoding.y)) {
+    // y is continuous but x is not
+    return 'vertical';
+  } else {
+    // Neither x nor y is continuous.
+    throw new Error('Need a valid continuous axis for errorbars');
+  }
+}
+
+
+function errorBarContinousAxis(spec: GenericUnitSpec<Encoding<string>, ErrorBar | ErrorBarDef>, orient: Orient) {
```

Comment:
This method is exactly the same as boxplot's (except the expected `aggregate`)? 

If so, extract this method to `common.ts` and reuse.